### PR TITLE
Doom.cpp

### PR DESCRIPTION
**Pull request by AJenbo**
This gets doom.cpp bin exact. I also lightly touched on UseItem(), mostly to better understand what triggers the quest and use a few consts.

Aiditionally this adds a .editorconf: https://editorconfig.org/ this atleast prevents mixed indentation untill we can setup a propper autoformatter like clang-format.

### CODE REVIEW

File: Source/doom.cpp [hunk 1]
```diff
@@ -10,13 +10,11 @@ int DoomQuestState; // idb
 
 int __cdecl doom_get_frame_from_time()
 {
-	int result; // eax
+	if ( DoomQuestState == 36001 ) {
```
Comment (mewmew) [inline Source/doom.cpp]: So one hour to complete the quest? :)

File: Source/doom.cpp [hunk 2]
```diff
@@ -10,13 +10,11 @@ int DoomQuestState; // idb
...
+		return 31;
+	}
 
-	if ( DoomQuestState == 36001 )
-		result = 31;
-	else
-		result = DoomQuestState / 1200;
-	return result;
+	return DoomQuestState / 1200;
```
Comment (mewmew) [inline Source/doom.cpp]: And the map of the stars updates every two minutes. Would be fun to play with this quest enabled.

**Comment by AJenbo** [comment]
Note the failed build is just Travis Mac server failing to init, rerunning it should solve it.

File: Source/doom.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/doom.cpp]: Looks like it, kinda tough :)

**Comment by galaxyhaxz** [comment]
For reference, if you check the 1.00 debug release code for the movie is still intact, but not called anywhere:
```
void __cdecl doom_reset_state()
{
    if ( DoomQuestState <= 0 )
        DoomQuestState = 0;
}

void __cdecl doom_play_movie()
{
    if ( DoomQuestState < 36001 && ++DoomQuestState == 36001 )
    {
        PlayInGameMovie("gendata\\doom.smk");
        ++DoomQuestState;
    }
}
```

**Comment by seritools** [comment]
Changed the editorconfig to match the CRLF line ending of the files, the rest LGTM :)

**Pull request merged**